scheduler/reconcile_test.go

```diff
@@ -1063,6 +1103,126 @@ func TestReconciler_LostNode_PreventRescheduleOnLost(t *testing.T) {
 	}
 }
 
+func TestReconciler_InformationalAllocs(t *testing.T) {
+	disabledReschedulePolicy := &structs.ReschedulePolicy{
+		Attempts:  0,
+		Unlimited: false,
+	}
+
+	ci.Parallel(t)
+	now := time.Now()
+
+	testCases := []struct {
+		name             string
+		count            int
+		stoppedCount     int
+		failedCount      int
+		reschedulePolicy *structs.ReschedulePolicy
+		expectPlace      int
+		expectStop       int
+		expectIgnore     int
+	}{
+		{
+			name:             "Count 3, 2 allocs failed, 1 stopped, no reschedule",
+			count:            3,
+			stoppedCount:     1,
+			failedCount:      2,
+			reschedulePolicy: disabledReschedulePolicy,
+			expectPlace:      2,
+			expectStop:       1,
+			expectIgnore:     1,
+		},
+		{
+			name:         "Count 1, 1 alloc failed, 1 stopped, reschedule",
+			count:        1,
+			stoppedCount: 1,
+			failedCount:  1,
+			reschedulePolicy: &structs.ReschedulePolicy{
+				Attempts: 1,
+			},
+			expectPlace:  1,
+			expectStop:   2,
+			expectIgnore: 0,
+		},
+		{
+			name:             "Count 2, no allocs failed, 2 stopped, no reschedule",
+			count:            2,
+			stoppedCount:     2,
+			failedCount:      0,
+			reschedulePolicy: disabledReschedulePolicy,
+			expectPlace:      2,
+			expectStop:       1,
+			expectIgnore:     0,
```

Comment:
I'm not a 100% confident about these. Intuitively, I would expect 2 allocs to place, 0 to ignore and 0 to stop. This might have to do with what nodes are available in this case, I will look into this and do some additional manual testing to be sure that we're setting alloc desired status and client status correctly.